### PR TITLE
Subject prefix - new PR

### PR DESCRIPTION
**Pull request by Nunien**
it replace this pr https://github.com/fli-iam/shanoir-ng/pull/1161

### CODE REVIEW

File: shanoir-ng-front/src/app/subjects/subject/subject.component.ts [hunk 1]
```diff
@@ -222,9 +222,9 @@ export class SubjectComponent extends EntityComponent<Subject> implements OnInit
 
     public toggleAnonymised() {
         if (this.isAlreadyAnonymized && this.subjectNamePrefix) {
-            this.subject.name = this.subjectNamePrefix;
-        } else if (!this.isAlreadyAnonymized && this.subjectNamePrefix && this.dicomPatientName) {
             this.subject.name = this.subjectNamePrefix + this.dicomPatientName + '-';
```
Comment (jcomedouteau) [inline shanoir-ng-front/src/app/subjects/subject/subject.component.ts:225]: This adds a "-" at the end of the subject name:
image

Is it on purpose ? If it's already anonymised, ha may want to let the name as is.

**Comment by jcomedouteau** [comment]
If no prefix is defined, the default study prefix is the study name. Did we agree to force it everytime ?

**Comment by jcomedouteau** [comment]
And a last thing by my fault, but changing the prefix does not trigger the form validity update.

**Comment by Nunien** [comment]
Thanks @jcomedouteau for your feedback, I have made made some changes.

**Review by jcomedouteau — APPROVED** [review]
Ok for me ! Maybe we can validate the logic with Elise and @michaelkain !

**Comment by jcomedouteau** [comment]
@Nunien you can merge this one :)

**Pull request merged**